extensions/barrage/src/main/java/io/deephaven/extensions/barrage/util/BarrageStreamReader.java

```diff
@@ -282,7 +290,19 @@ public BarrageMessage safelyParseFrom(final StreamReaderOptions options,
             }
 
             if (header != null && header.headerType() == MessageHeader.Schema) {
-                // there is no body and our clients do not want to see schema messages
+                // there is no body and our clients do not want to see schema messages, consume the schema so that we
+                // can read the following messages and return null.
+                ByteBuffer original = header.getByteBuffer();
+                ByteBuffer copy = ByteBuffer.allocate(original.remaining()).put(original).rewind();
+                Schema schema = new Schema();
+                Message.getRootAsMessage(copy).header(schema);
+                header.header(schema);
```

Comment:
If possible I would like it more obvious why we need to copy here. (e.g. a comment related to what references get leaked)

Reply:
I'm not 100% that we need to copy in this case - technically it appears no since line 87 does a ByteBuffer.wrap(). Instead this is an attempt to be defensive in case a future impl is reading from a slice/etc of the ByteBuffer that came in over the wire.